packages/fxa-settings/src/lib/auth-errors/auth-errors.ts

```diff
@@ -645,64 +646,74 @@ export const getLocalizedErrorMessage = (
-    if (
-      error.retryAfterLocalized &&
-      error.errno === AuthUiErrors.THROTTLED.errno
-    ) {
-      // For throttling errors where a localized retry after value is provided
-      localizedError = ftlMsgResolver.getMsg(
-        composeAuthUiErrorTranslationId(error),
-        AuthUiErrorNos[error.errno].message,
-        { retryAfter: error.retryAfterLocalized }
-      );
-    } else if (error.errno === AuthUiErrors.THROTTLED.errno) {
-      // For throttling errors where a localized retry after value is not available
-      localizedError = ftlMsgResolver.getMsg(
-        'auth-error-114-generic',
-        AuthUiErrorNos[114].message
-      );
-    } else {
-      // for all other recognized auth UI errors
-      localizedError = ftlMsgResolver.getMsg(
-        composeAuthUiErrorTranslationId(error),
-        AuthUiErrorNos[error.errno].message
-      );
+  if (error.errno) {
+    if (AuthUiErrorNos[error.errno]) {
+      if (
+        error.retryAfterLocalized &&
+        error.errno === AuthUiErrors.THROTTLED.errno
+      ) {
+        // For throttling errors where a localized retry after value is provided
+        return ftlMsgResolver.getMsg(
+          getErrorFtlId(error),
+          AuthUiErrorNos[error.errno].message,
+          { retryAfter: error.retryAfterLocalized }
+        );
+      } else if (error.errno === AuthUiErrors.THROTTLED.errno) {
+        // For throttling errors where a localized retry after value is not available
+        return ftlMsgResolver.getMsg(
+          'auth-error-114-generic',
+          AuthUiErrorNos[114].message
+        );
+      } else {
+        // for all other recognized auth UI errors
+        return ftlMsgResolver.getMsg(
+          getErrorFtlId(error),
+          AuthUiErrorNos[error.errno].message
+        );
+      }
     }
-  } else {
-    const unexpectedError = AuthUiErrors.UNEXPECTED_ERROR;
-    localizedError = ftlMsgResolver.getMsg(
-      composeAuthUiErrorTranslationId(unexpectedError),
-      unexpectedError.message
+    const oAuthError = Object.values(OAUTH_ERRORS).find(
+      (oAuthErr) => error.errno === oAuthErr.errno
     );
+    if (oAuthError) {
+      return ftlMsgResolver.getMsg(
+        getErrorFtlId(oAuthError),
+        oAuthError.message
+      );
+    }
   }
-  return localizedError;
+
+  const unexpectedError = AuthUiErrors.UNEXPECTED_ERROR;
+  return ftlMsgResolver.getMsg(
+    getErrorFtlId(unexpectedError),
+    unexpectedError.message
+  );
 };
 
-export const composeAuthUiErrorTranslationId = (err: {
-  errno?: number;
-  message?: string;
-}) => {
-  /* all of these checks for fields/values being present look a little wonky, but allow us to work with
-   * the AuthUiError type, which has all optional fields. Previously this wasn't an issue bc we didn't use
-   * a utility.
-   */
+export const getErrorFtlId = (err: { errno?: number; message?: string }) => {
+  if (err.errno) {
+    if (err.errno in AuthUiErrorNos) {
```

Comment:
Hmm, this is a good callout. Somehow I missed that they can overlap.

I'll make a comment about this in the code, and make it part of the issue to localize these error messages in `OAuthDataError` since we're going to have to fix this there.